### PR TITLE
Op: Sub, Pow and Sqrt

### PR DESCRIPTION
**Pull request by briangrifiin**
* Subtraction: https://github.com/onnx/onnx/blob/master/docs/Operators.md#Sub
* Power: https://github.com/onnx/onnx/blob/master/docs/Operators.md#pow
* Square root: https://github.com/onnx/onnx/blob/master/docs/Operators.md#sqrt

### CODE REVIEW

File: tests/test_cmd.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def cmd_convert(model_name):
 def test_cmd_convert():
     MODEL_LIST = (
         'abs.float32',
+        'abs-sqrt.float32',
```
Comment (jackwish) [inline tests/test_cmd.py:22]: In general, we don't need to add more operator tests to the command line tool test, but won't block the PR merging.

File: tflite2onnx/op/binary.py [hunk 2]
```diff
@@ -99,10 +102,11 @@ def parse(self):
...
         # options
         op_opt = op.BuiltinOptions()
-        option = self.OptionMapping[opcode]()
-        option.Init(op_opt.Bytes, op_opt.Pos)
+        if opcode in self.OptionMapping:
+            option = self.OptionMapping[opcode]()
+            option.Init(op_opt.Bytes, op_opt.Pos)
 
-        handleFusedActivation(self, option, ot)
+            handleFusedActivation(self, option, ot)
```
Comment (jackwish) [inline tflite2onnx/op/binary.py:109]: Thanks for fixing this, I don't even know there is an issue here.

File: tests/test_ops.py [hunk 3]
```diff
@@ -85,6 +85,9 @@ def test_ops_layout_transparent():
         'relu.float32',
         'prelu.float32',
         'sigmoid.float32',
+        'sub.float32',
+        'abs-sqrt.float32',
```
Comment (jackwish) [inline tests/test_ops.py:89]: I am curious about if it's possible to have `sqrt` and `power` models by themselves. Dedicated models containing only the specific operator could be helpful when we debugging.

**Review by jackwish — CHANGES_REQUESTED** [review]
Hi @briangrifiin thanks for your active contribution!

This PR looks good to me in general. I am just curious about if it's possible to have `sqrt` and `power` models by themselves - dedicated models containing only the specific operator could be helpful when we debugging.

Btw, as you may see in [the **Going further** section of the operator enabling guide](https://github.com/jackwish/tflite2onnx/blob/master/docs/how-to-enable-new-operator.md#going-further), you can add the newly enabled operators to [the operator support list](https://github.com/jackwish/tflite2onnx/blob/master/docs/operator-support-status.md). It will be helpful if we'd like to know which operators are supported currently.

Thanks!

File: tests/test_cmd.py (hunk 1 above)
Comment (briangrifiin) [reply tests/test_cmd.py]: https://github.com/jackwish/tflite2onnx/pull/37/commits/90c439701cc1ea57daf7cf540d8b0033410eb565#diff-60497cc703a3a9baddaf30a3384a7774b89bc52007e28c34828758f055fcd72cL22

File: tflite2onnx/op/binary.py (hunk 2 above)
Comment (briangrifiin) [reply tflite2onnx/op/binary.py:109]: Power is a binary operation, but it doesn't have any options. Therefore, I added this check.

File: tests/test_ops.py (hunk 3 above)
Comment (briangrifiin) [reply tests/test_ops.py:89]: `m.genInput()` in `end2end_test` generates random numbers. However, square root from negative real numbers does not produce real number (results in NaN instead), thus I had to either use complex data type or get absolute values before applying square root.

Same with the power. If some negative power is applied to zero, that has the same effect as division by zero. I have also had some troubles with large exponentiation, like Pow(100, 1000), so I decided to clamp input values with ReLU6.

If you happen to know any other option to limit input values, please let me know.

Comment (jackwish) [reply tests/test_ops.py:89]: Oh, we need to add options to [`shrub`](https://github.com/jackwish/shrub/blob/master/shrub/network.py#L99). But that will be a TODO. Thanks for the explanation.

**Review by jackwish — APPROVED** [review]

**Comment by jackwish** [comment]
@briangrifiin This is now merged, thanks for your contribution!

**Pull request merged**